### PR TITLE
Issue #5107: ICU Data Scripts

### PR DESCRIPTION
**Pull request by hawkfish**
Import and update the scripts for building the ICU data sets.

### CODE REVIEW

File: extension/icu/scripts/inline-data.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+# encode the data into the stubdata file
+result_text = ""
+first = True
+for byte in contents:
+	if first:
+		result_text += str(byte)
+	else:
+		result_text += "," + str(byte)
+	first = False
```
Comment (Mause) [inline extension/icu/scripts/inline-data.py:13]:
```suggestion
result_text = ",".join(map(str, contents))
```

File: extension/icu/scripts/makedata.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
+#rm -rf build
+mkdir build
+pushd build
```
Comment (Mytherin) [inline extension/icu/scripts/makedata.sh:3]: Maybe add `set -e` if we are going to be using an evil shell script?

File: extension/icu/scripts/makedata.sh [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+# download ICU 66
+wget -nc https://github.com/unicode-org/icu/archive/refs/tags/release-66-1.zip
+unzip -o release-66-1.zip
+
+# download ICU 71 (replace with latest version)
+wget -nc https://github.com/unicode-org/icu/archive/refs/tags/release-71-1.zip
```
Comment (Mytherin) [inline extension/icu/scripts/makedata.sh:10]: Should the version not be parameterized?

**Review by Mytherin — COMMENTED** [review]
Thanks! LGTM

File: extension/icu/scripts/inline-data.py (hunk 1 above)
Comment (hawkfish) [reply extension/icu/scripts/inline-data.py:13]: Yeah that was bothering me too (scars from an old Python project where string concatenation was a performance bottleneck until we figured this out!) I was just trying to get @Mytherin 's existing scripts in with a minimum of modification to start. I'd also like to break up the lines to make diffing less stressful.

File: extension/icu/scripts/makedata.sh (hunk 3 above)
Comment (hawkfish) [reply extension/icu/scripts/makedata.sh:10]: I was wondering about that but on the flip side its nice to have the correct version checked in. If I parameterised it I'd have to add another script.

Comment (hawkfish) [reply extension/icu/scripts/makedata.sh:11]: For the next PR.  Plus we might want to separate the TZ drops from the ICU versions (they do that for Java).

**Pull request merged**